### PR TITLE
Handle a connection abort error on shutdown

### PR DESCRIPTION
**Pull request by blink1073**
I was getting the following traceback on shutdown, which is now avoided:

```python
*** End of MainWindow setup ***
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/Users/silvester/anaconda/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/Users/silvester/anaconda/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/silvester/workspace/spyder/spyderlib/spyder.py", line 2641, in start_open_files_server
    req, dummy = self.open_files_server.accept()
  File "/Users/silvester/anaconda/lib/python3.4/socket.py", line 187, in accept
    fd, addr = self._accept()
ConnectionAbortedError: [Errno 53] Software caused connection abort
```

### CODE REVIEW

File: spyderlib/spyder.py [hunk 1]
```diff
@@ -2646,6 +2646,9 @@ def start_open_files_server(self):
                 # To avoid a traceback after closing on Windows
                 if e.args[0] == eintr:
                     continue
+                # handle a connection abort on close error
```
Comment (blink1073) [inline spyderlib/spyder.py]: This is the relevant change.

File: spyderlib/spyder.py [hunk 2]
```diff
@@ -358,7 +358,7 @@ def __init__(self, options=None):
         # Tour  # TODO: Should I consider it a plugin?? or?
         self.tour = None
         self.tours_available = None
-
```
Comment (ccordoba12) [inline spyderlib/spyder.py]: Please revert blanks deletion :-)

File: spyderlib/spyder.py [hunk 3]
```diff
@@ -2646,6 +2646,9 @@ def start_open_files_server(self):
                 # To avoid a traceback after closing on Windows
                 if e.args[0] == eintr:
                     continue
+                # handle a connection abort on close error
+                if e.args[0] == errno.ECONNABORTED:
+                    return
```
Comment (ccordoba12) [inline spyderlib/spyder.py]: Does this fix the closing error on Windows too?

**Comment by blink1073** [comment]
Fixed ws and added a fix for Windows.

File: spyderlib/spyder.py [hunk 4]
```diff
@@ -2646,6 +2646,9 @@ def start_open_files_server(self):
                 # To avoid a traceback after closing on Windows
                 if e.args[0] == eintr:
                     continue
+                # handle a connection abort on close error
+                if e.args[0] in [errno.ECONNABORTED, errno.ENOTSOCK]:
+                    return
```
Comment (ccordoba12) [inline spyderlib/spyder.py]: Why here `return` and not `continue`? I say it because I tried to fix this error in commit e18294fbba9a91c815a234da8df16b18824e2bb0 but I thought the right thing to do was to use `continue`. Later I had to revert that one because complained that Spyder took up to half an hour to start or close!! :-)

Comment (ccordoba12) [inline spyderlib/spyder.py]: I also forgot to mention that `errno.ENOTSOCK` is not available on Windows. See my referenced commit for the right error on Win.

Comment (blink1073) [inline spyderlib/spyder.py]: I used `return` because it only happened at shutdown.

Comment (blink1073) [inline spyderlib/spyder.py]: It is so frustrating when they do that in the standard library.

File: spyderlib/spyder.py [hunk 5]
```diff
@@ -2646,6 +2646,11 @@ def start_open_files_server(self):
                 # To avoid a traceback after closing on Windows
                 if e.args[0] == eintr:
                     continue
+                # handle a connection abort on close error
+                enotsock = (errno.WSAENOTSOCK if os.name == 'nt'
+                            else errno.ENOTSOCK)
+                if e.args[0] == enotsock:
```
Comment (ccordoba12) [inline spyderlib/spyder.py]: Why did you remove `ECONNABORTED`?

Comment (ccordoba12) [inline spyderlib/spyder.py]: At the end this is essentially what I added before, but with `return` instead of `continue` ;-)

**Comment by blink1073** [comment]
Whoops, fixed.

**Comment by goanpeca** [comment]
@blink1073 squashing into a single commit?

**Comment by blink1073** [comment]
Done.

**Comment by ccordoba12** [comment]
Ok, let's hope this fixes this problem once and for all :-)

Merging!

**Comment by blink1073** [comment]
Much nicer without that pesky error on shutdown!

**Comment by ccordoba12** [comment]
Yes, absolutely!! I've been trying to fix this bug for months without success :-)
